### PR TITLE
ETL Plugin Details

### PR DESCRIPTION
**Pull request by awholegunch**
Updates the ETL Plugin Details, for both the rst and Java source files.

Adds links to the table of Plugins, so that you jump to them directly.

Build passed as SDK build on a local machine.

Built docs at: http://builds.cask.co/browse/CDAP-DDB7-2

### CODE REVIEW

File: cdap-docs/included-applications/source/etl/plugins/batchsinks/elasticsearch.rst [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+exist, it will be created.
+
+**idField:** The field that will determine the id for the document. It should match a fieldname
+in the structured record of the input.
+
+**hostname:** The hostname and port for the Elasticsearch server; such as localhost:9200.
+
+.. rubric:: Example
+
+
```
Comment (albertshau) [inline cdap-docs/included-applications/source/etl/plugins/batchsinks/elasticsearch.rst]: we can take the example from https://github.com/caskdata/cdap-plugins/tree/develop/elasticsearch-plugins.  Not sure why we have elastic search in 2 places though. @sreevatsanraman, @nitinmotgi  do you know why?

File: cdap-docs/included-applications/source/etl/plugins/batchsinks/tpfsparquet.rst [hunk 2]
```diff
@@ -1,47 +1,51 @@
...
+this setting is only used if filepathformat is not null.
 
-.. rubric:: Example
+.. rubric:: example
 
 ::
 
   {
-    "name": "TPFSAvro",
+    "name": "tpfsavro",
```
Comment (albertshau) [inline cdap-docs/included-applications/source/etl/plugins/batchsinks/tpfsparquet.rst]: these cannot be changed, it is important they match what is in the Java code.

File: cdap-docs/included-applications/source/etl/plugins/batchsinks/tpfsparquet.rst [hunk 3]
```diff
@@ -1,47 +1,51 @@
...
     "properties": {
       "name": "users",
-      "filePathFormat": "yyyy-MM-dd/HH-mm",
-      "timeZone": "America/Los_Angeles",
+      "filepathformat": "yyyy-mm-dd/hh-mm",
```
Comment (albertshau) [inline cdap-docs/included-applications/source/etl/plugins/batchsinks/tpfsparquet.rst]: don't change these capitalization is important in both the key and the value. MM means month while mm means minute (http://docs.oracle.com/javase/7/docs/api/java/text/SimpleDateFormat.html)

File: cdap-docs/included-applications/source/etl/plugins/batchsinks/tpfsparquet.rst [hunk 4]
```diff
@@ -1,47 +1,51 @@
...
+      "timezone": "america/los_angeles",
```
Comment (albertshau) [inline cdap-docs/included-applications/source/etl/plugins/batchsinks/tpfsparquet.rst]: same here, keep the cases

File: cdap-docs/included-applications/source/etl/plugins/batchsources/elasticsearch.rst [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+**index:** The name of the index to query.
+
+**query:** The query to use to import data from the specified index. See Elasticsearch for
+query examples.
+
+**schema:** The schema or mapping of the data in Elasticsearch.
+
+**type:** The name of the type where the data is stored.
+
+.. rubric:: Example
```
Comment (albertshau) [inline cdap-docs/included-applications/source/etl/plugins/batchsources/elasticsearch.rst]: similar comment about taking the example from https://github.com/caskdata/cdap-plugins/tree/develop/elasticsearch-plugins

File: cdap-docs/included-applications/source/etl/plugins/realtimesinks/elasticsearch.rst [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+exist, it will be created.
+
+**idField:** The field that will determine the id for the document. It should match a fieldname
+in the structured record of the input.
+
+**hostname:** The hostname and port for the Elasticsearch server; such as localhost:9200.
+
+.. rubric:: Example
+
+
```
Comment (albertshau) [inline cdap-docs/included-applications/source/etl/plugins/realtimesinks/elasticsearch.rst]: similar comment about example from https://github.com/caskdata/cdap-plugins/tree/develop/elasticsearch-plugins

File: cdap-docs/included-applications/source/etl/plugins/batchsinks/tpfsparquet.rst (hunk 2 above)
Comment (albertshau) [inline cdap-docs/included-applications/source/etl/plugins/batchsinks/tpfsparquet.rst]: actually, this was wrong to begin with. Should be TPFSParquet

File: cdap-docs/included-applications/source/etl/plugins/batchsinks/tpfsparquet.rst (hunk 3 above)
Comment (awholegunch) [inline cdap-docs/included-applications/source/etl/plugins/batchsinks/tpfsparquet.rst]: See above.

File: cdap-docs/included-applications/source/etl/plugins/batchsinks/tpfsparquet.rst (hunk 2 above)
Comment (awholegunch) [inline cdap-docs/included-applications/source/etl/plugins/batchsinks/tpfsparquet.rst]: This was an accident. I was changing the case of some text (to create the link at the top of the page) and it looks I didn't use my editor correctly, and set the entire file to lowercase. Everything is wrong. I'll fix it and you can double-check it.

Comment (awholegunch) [inline cdap-docs/included-applications/source/etl/plugins/batchsinks/tpfsparquet.rst]: Fixed.

File: cdap-docs/included-applications/source/etl/plugins/batchsinks/tpfsparquet.rst (hunk 4 above)
Comment (awholegunch) [inline cdap-docs/included-applications/source/etl/plugins/batchsinks/tpfsparquet.rst]: Fixed.

File: cdap-docs/included-applications/source/etl/plugins/batchsinks/elasticsearch.rst (hunk 1 above)
Comment (awholegunch) [inline cdap-docs/included-applications/source/etl/plugins/batchsinks/elasticsearch.rst]: What did you mean by "elasticsearch in two places"? We have:
- Batch Sinks: Elasticsearch
- Batch Sources: Elasticsearch
- Real-time Sinks: Elasticsearch
Is this not correct?

**Comment by awholegunch** [comment]
@albertshau I've updated and fixed things as per your comments. Please take another look.

**Comment by albertshau** [comment]
lgtm after conflicts are fixed

**Comment by awholegunch** [comment]
I'm rebuilding it so that it's easier to see the final product:

- [Creating Custom ETL plugins](http://builds.cask.co/artifact/CDAP-DDB7/shared/build-3/Docs-HTML/3.2.0-SNAPSHOT/en/included-applications/etl/custom.html)

**Comment by awholegunch** [comment]
@albertshau I've started a new build; please take a look at the above link when it's done and let me know if you're fine with the titles and arrangement of that page. I think the merge has resolved correctly...

**Comment by albertshau** [comment]
:+1:

**Pull request merged**